### PR TITLE
[Azure Pipelines] Upgrade to macOS 11 (Big Sur)

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by foolip** [comment]
This did not go as smoothly as I had hoped. There have apparently been changes to the security model for installing certificates, leading to this failure:

> SecTrustSettingsSetTrustSettings: errSecInternalComponent

@gsnedders do you know if it's possible to work around this in any way?

**Comment by gsnedders** [comment]
I'm pretty sure the problem is that `/Library/Keychains/System.keychain` is on a non-writeable volume as of Big Sur; adding it to the user keychain is probably the right solution here.

**Comment by foolip** [comment]
Thanks, I'll look into that!

**Comment by foolip** [comment]
@gsnedders I tried some things but no luck. The latest is this:
```sh
KEYCHAIN=$(security default-keychain -d user)
echo $KEYCHAIN
KEYCHAIN=$(eval echo $KEYCHAIN)
echo $KEYCHAIN
sudo security add-trusted-cert -r trustRoot -k "$KEYCHAIN" tools/certs/cacert.pem
```

The `echo`s confirmed the path is /Users/runner/Library/Keychains/login.keychain-db. In other words, trying to add the cert to the user keychain but using `sudo` to avoid the password prompt. But that doesn't work, the error is still "SecTrustSettingsSetTrustSettings: errSecInternalComponent".

This feels very similar to past issues of `sudo safaridriver --enable` not working. Do you have some ideas for other things to try? Perhaps it's possible to tell Safari (STP) which keychain (or cert directly) to use in some other way, as command line arguments or via WebDriver?

**Comment by stephenmcgruer** [comment]
On my local macbook, doing:

```
sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain tools/certs/cacert.pem
OR
sudo security add-trusted-cert -r trustRoot -k /Users/smcgruer/Library/Keychains/login.keychain-db tools/certs/cacert.pem
```

Both prompt me for my sudo password on the cmdline, **and** then open a UX dialog explaining I am making changes to the 'System Certificate Trust Settings' and that I need to enter my password to allow it...

https://github.com/travis-ci/travis-ci/issues/5789 mentions this (long before Big Sur?). It ends up pointing to https://www.jviotti.com/2016/03/16/how-to-code-sign-os-x-electron-apps-in-travis-ci.html which creates its own keychain instead of using the system one. That doesn't necessarily feel like the right solution though.

Another forum post I found, with a lot of sysadmins complaining about this change in Big Sur :D -  https://developer.apple.com/forums/thread/671582

Some folks have also noted that you can 'echo' the password into the cmd, but this requires putting it in the clear. (And for azure pipelines we likely don't have that password anyway...)

**Comment by foolip** [comment]
Thanks for digging @stephenmcgruer! Indeed we don't have the password in this case, or at least when I was dealing with the `safaridriver --enable` issue I couldn't get it from the Azure Pipelines team.

**Comment by gsnedders** [comment]
The only way to do add certificates without explicit user consent is via disabling SIP.

**Comment by stephenmcgruer** [comment]
Relevant documentation from Apple: https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection (which feels fairly infeasible...)

**Comment by thejohnjansen** [comment]
I do not think it is possible. We haven't spent a lot of time investigating work-arounds, but the design implies this is not possible.

**Comment by gsnedders** [comment]
@foolip this appears to currently be failing due to chromedriver on both Linux and macOS, `unknown error (500): unknown error: cannot find Chrome binary`, which is pretty unfortunate

https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=73539 has a full run of STP underway, and hopefully that comes out looking okay

**Comment by foolip** [comment]
That run completed: https://wpt.fyi/results/?run_id=5714547575881728

**Comment by foolip** [comment]
https://wpt.fyi/results/?run_id=5638166280142848 is the closest run to compare to, so:
https://wpt.fyi/results/?diff&filter=ADC&run_id=5638166280142848&run_id=5714547575881728

Overall, definitely more fixes than regressions.

@gsnedders I guess only passing the CI now blocks this? Is "cannot find Chrome binary" also happening on master? I'll revive my probing PR to see...

File: tools/ci/azure/install_certs.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 steps:
 - script: |
     # https://github.com/web-platform-tests/results-collection/blob/master/src/scripts/trust-root-ca.sh
-    sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain tools/certs/cacert.pem
+    # only run this on macOS < 11
```
Comment (jgraham) [inline tools/ci/azure/install_certs.yml:4]: Couldn't we just remove this whole step?

**Review by jgraham — APPROVED** [review]

File: tools/ci/azure/install_certs.yml (hunk 1 above)
Comment (gsnedders) [reply tools/ci/azure/install_certs.yml:4]: we're still running Safari Stable on macOS 10.15, and we don't have `acceptInsecureCerts` support in stable

**Comment by foolip** [comment]
Azure Pipelines and Taskcluster were happy in https://github.com/web-platform-tests/wpt/pull/28759, so failures here are probably introduced by the changes.

**Review by foolip — COMMENTED** [review]
I can't approve since I started the PR, but current changes (sans the `./wpt test-jobs --all` bits) look good!

**Review by Marcobanderas — APPROVED** [review]

**Pull request merged**